### DISCUSSION TITLE
Why AMC Pruner use the training sub-set as the validation set?

### DISCUSSION THREAD

**Discussion by twmht** [microsoft/nni#3999] [General]
Hi,

I recently read the code about AMC Pruner, but found out that it splits training set into sub training set and a validation set (https://github.com/microsoft/nni/blob/master/examples/model_compress/pruning/amc/amc_search.py#L73).

But why not use the original validation set?

In the white paper they said the reason is to prevent from overfitting, But I can't understand and get convinced.

Any idea?

**Comment by linbinskn** [comment]
Yes, splitting training set into `train_loader` and `val_loader` is to be aligned with AMC paper. The AMC author uses 5k images from cifar10 **training images** and 3k images from imagenet **training images** as validation dataset. You can check the evaluation part of that paper.
